doc/CMakeLists.txt

```diff
@@ -46,65 +46,65 @@ IF(WITH_APIDOC)
     ENDIF(GENERATE_QHP)
 
     SET(DOXYGEN_INCLUDE_PATH
-      ${CMAKE_SOURCE_DIR}/src/core
+      ${CMAKE_CURRENT_SOURCE_DIR}/../src/core
     )
     STRING(REPLACE ";" " " DOXYGEN_INCLUDE_PATH "${DOXYGEN_INCLUDE_PATH}")
 
     SET(DOXYGEN_INPUT
-      ${CMAKE_SOURCE_DIR}/doc
-      ${CMAKE_SOURCE_DIR}/src/core
-      ${CMAKE_SOURCE_DIR}/src/core/annotations
-      ${CMAKE_SOURCE_DIR}/src/core/auth
-      ${CMAKE_SOURCE_DIR}/src/core/composer
-      ${CMAKE_SOURCE_DIR}/src/core/diagram
-      ${CMAKE_SOURCE_DIR}/src/core/dxf
-      ${CMAKE_SOURCE_DIR}/src/core/effects
-      ${CMAKE_SOURCE_DIR}/src/core/fieldformatter
-      ${CMAKE_SOURCE_DIR}/src/core/geometry
-      ${CMAKE_SOURCE_DIR}/src/core/gps
-      ${CMAKE_SOURCE_DIR}/src/core/layertree
-      ${CMAKE_SOURCE_DIR}/src/core/metadata
-      ${CMAKE_SOURCE_DIR}/src/core/pal
-      ${CMAKE_SOURCE_DIR}/src/core/processing
-      ${CMAKE_SOURCE_DIR}/src/core/providers
-      ${CMAKE_SOURCE_DIR}/src/core/providers/memory
-      ${CMAKE_SOURCE_DIR}/src/core/raster
-      ${CMAKE_SOURCE_DIR}/src/core/scalebar
-      ${CMAKE_SOURCE_DIR}/src/core/symbology-ng
-      ${CMAKE_SOURCE_DIR}/src/gui
-      ${CMAKE_SOURCE_DIR}/src/gui/auth
-      ${CMAKE_SOURCE_DIR}/src/gui/attributetable
-      ${CMAKE_SOURCE_DIR}/src/gui/editorwidgets
-      ${CMAKE_SOURCE_DIR}/src/gui/editorwidgets/core
-      ${CMAKE_SOURCE_DIR}/src/gui/effects
-      ${CMAKE_SOURCE_DIR}/src/gui/layertree
-      ${CMAKE_SOURCE_DIR}/src/gui/locator
-      ${CMAKE_SOURCE_DIR}/src/gui/raster
-      ${CMAKE_SOURCE_DIR}/src/gui/symbology-ng
-      ${CMAKE_SOURCE_DIR}/src/analysis
-      ${CMAKE_SOURCE_DIR}/src/analysis/interpolation
-      ${CMAKE_SOURCE_DIR}/src/analysis/network
-      ${CMAKE_SOURCE_DIR}/src/analysis/openstreetmap
-      ${CMAKE_SOURCE_DIR}/src/analysis/raster
-      ${CMAKE_SOURCE_DIR}/src/analysis/vector
-      ${CMAKE_SOURCE_DIR}/src/plugins
+      ${CMAKE_CURRENT_SOURCE_DIR}/../doc
```

Comment:
It's kind of ok here.

But you soon go 3 levels up and then it requires a lot of brainpower to determine, where the target is (`../../core/`: was it the core directory in src/, python/, tests/? etc.).
Another drawback is, that when adding a new folder (e.g. the subfolder `src/core/expressions`) you can't just copy-paste this between different CMakeLists but you have to manually check for each and every single CMakeLists file the relative path.

Personally, I find these relative paths in CMakeLists a PITA to maintain and would prefer to switch it all to "relative to source root". And move this CMakeLists file here also to the root.

So I wonder if there is a possibility to move the logic from this CMakeLists file here to the main one or move all code related to this one to this subdirectory.